File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        char_dict = dict()
        already_list = []
        for idx, c in enumerate(s):
            if c in already_list: 
                char_dict[c] = -1
            else:
                char_dict[c] = idx
                already_list.append(c)
        
        min_loc = 987654321
        for key, idx in char_dict.items():
            if idx != -1:
                if min_loc > idx:
                    min_loc = idx
        return min_loc if min_loc != 987654321 else -1

        

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> freq(26,0);
        vector<int> loc(26,-1);
        for (int i = 0; i < s.length(); i++) {
            char c = s[i];
            freq[c-'a']++;
            if (loc[c-'a'] == -1) loc[c-'a'] = i;
        }
        int ans = s.length(); 
        for (int j = 0; j < 26; j++) {
            if (freq[j] == 1) {
                ans = min(ans, loc[j]);
            }
        }
        if (ans != s.length()) return ans;
        return -1;
    }
};